homeshick.sh

```diff
@@ -1,12 +1,15 @@
 # This script should be sourced in the context of your shell like so:
-# source $HOME/.homeshick/repos/.homeshick/homeshick.sh
+# source homeshick.sh
 # Once the homeshick() function is defined, you can type
 # "homeshick cd CASTLE" to enter a castle.
 
-function homeshick() {
-	if [ "$1" = "cd" ] && [ -n "$2" ]; then
-		cd "$HOME/.homesick/repos/$2"
+__DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+eval "function homeshick() {
```

Comment:
The purpose is to substitute `__DIR` with the installation location of homeshick at function _definition_ time (i.e., when the script is sourced). All other variable substitutions occur at function _execution_ time (because they are escaped in the argument to eval). The result is that the shell sees a function definition like this:
```
$ type homeshick
homeshick is a function
homeshick () 
{ 
    if [ "$1" = "cd" ] && [ -n "$2" ]; then
        cd "$HOME/.homesick/repos/$2";
    else
        HOMESHICK_DIR="/usr/share/homeshick" "/usr/share/homeshick/bin/homeshick" "$@";
    fi
}
```